simulation.c

```diff
@@ -419,6 +421,101 @@ void stepSimulation(void)
    }
 }
 
+void stepGamemode(int timeremain)
+{
+   int ip;
+   //tick approx once per second:
+   if( !(timeremain % 64) )
+   {
+      if(conf.gametype != CONFIG_GAMETYPE_DEFAULT)
+      {
+         switch(conf.gametype)
+         {
+            case CONFIG_GAMETYPE_PREFERED_TARGET:
+            {
+               static int currentPreferedTarget;
+               char choosenew=1;
+               //Find active player that is prefered target
+               for(ip = 0; ip < conf.maxPlayers; ++ip)
+               {
+                   SimPlayer* p = &(player[ip]);
+                   if(p->active && p->extrapoints)
+                   {
+                      choosenew=0;
+                      break;
+                   }
+               }
+               
+               if(choosenew)
+               {
+                  //no prefered target. Choose a new one:
+                  ip=(currentPreferedTarget+1) % conf.maxPlayers;
+                  for(ip=(currentPreferedTarget+1) % conf.maxPlayers;
+                      ip != currentPreferedTarget; ip=(ip+1)% conf.maxPlayers)
+                  {
+                      SimPlayer* p = &(player[ip]);
+                      if(p->active)
+                      {
+                         printf("%d is new prefered target\n",ip);
+                         p->extrapoints=1;
+                         currentPreferedTarget=ip;
+                         break;
+                      }
+                  }
+               }
+               break;
+            }
+            case CONFIG_GAMETYPE_KILL_OLDEST:
+            {
+               static int tickcount;
+               ++tickcount;
+               if(!(tickcount%64))
+               {
+                  for(ip = 0; ip < conf.maxPlayers; ++ip)
+                  {
+                      SimPlayer* p = &(player[ip]);
+                      if(p->active && p->extrapoints<3)
+                      {
+                         ++p->extrapoints;
+                         break;
+                      }
+                  }
+                  tickcount=0;
+               }
+               break;
+            }
+            case CONFIG_GAMETYPE_KILL_BEST:
```

Comment:
If the number of kills is the same, than no one is choosen.
`if( p->kills ==  maxkills)
                      {
                         bestplayer=-1;
                      }`

But yes, taking the death into account sounds good to me. :)